test/dynamic_realm_test.dart

```diff
@@ -69,8 +69,117 @@ Future<void> main([List<String>? args]) async {
+    expect(actual.dynamic.get('nullableUuidProp'), expected.nullableUuidProp);
+
+    expect(actual.dynamic.get<int>('intProp'), expected.intProp);
+    expect(actual.dynamic.get('intProp'), expected.intProp);
+    expect(actual.dynamic.get<int?>('nullableIntProp'), expected.nullableIntProp);
+    expect(actual.dynamic.get('nullableIntProp'), expected.nullableIntProp);
+
+    dynamic actualDynamic = actual;
+    expect(actualDynamic.stringProp, expected.stringProp);
+    expect(actualDynamic.nullableStringProp, expected.nullableStringProp);
+    expect(actualDynamic.boolProp, expected.boolProp);
+    expect(actualDynamic.nullableBoolProp, expected.nullableBoolProp);
+    expect(actualDynamic.dateProp, expected.dateProp);
+    expect(actualDynamic.nullableDateProp, expected.nullableDateProp);
+    expect(actualDynamic.doubleProp, expected.doubleProp);
+    expect(actualDynamic.nullableDoubleProp, expected.nullableDoubleProp);
+    expect(actualDynamic.objectIdProp, expected.objectIdProp);
+    expect(actualDynamic.nullableObjectIdProp, expected.nullableObjectIdProp);
+    expect(actualDynamic.uuidProp, expected.uuidProp);
+    expect(actualDynamic.nullableUuidProp, expected.nullableUuidProp);
+    expect(actualDynamic.intProp, expected.intProp);
+    expect(actualDynamic.nullableIntProp, expected.nullableIntProp);
+  }
+
+  void _validateDynamicLists(RealmObject actual, AllCollections expected) {
+    expect(actual.dynamic.getList<String>('strings'), expected.strings);
+    expect(actual.dynamic.getList('strings'), expected.strings);
+
+    expect(actual.dynamic.getList<bool>('bools'), expected.bools);
+    expect(actual.dynamic.getList('bools'), expected.bools);
+
+    expect(actual.dynamic.getList<DateTime>('dates'), expected.dates);
+    expect(actual.dynamic.getList('dates'), expected.dates);
+
+    expect(actual.dynamic.getList<double>('doubles'), expected.doubles);
+    expect(actual.dynamic.getList('doubles'), expected.doubles);
+
+    expect(actual.dynamic.getList<ObjectId>('objectIds'), expected.objectIds);
+    expect(actual.dynamic.getList('objectIds'), expected.objectIds);
+
+    expect(actual.dynamic.getList<Uuid>('uuids'), expected.uuids);
+    expect(actual.dynamic.getList('uuids'), expected.uuids);
+
+    expect(actual.dynamic.getList<int>('ints'), expected.ints);
+    expect(actual.dynamic.getList('ints'), expected.ints);
+
+    dynamic actualDynamic = actual;
+    expect(actualDynamic.strings, expected.strings);
+    expect(actualDynamic.bools, expected.bools);
+    expect(actualDynamic.dates, expected.dates);
+    expect(actualDynamic.doubles, expected.doubles);
+    expect(actualDynamic.objectIds, expected.objectIds);
+    expect(actualDynamic.uuids, expected.uuids);
+    expect(actualDynamic.ints, expected.ints);
+  }
+
   for (var isDynamic in [true, false]) {
-    Realm getDynamicRealm(Realm original) {
+    Realm _getDynamicRealm(Realm original) {
```

Comment:
The changes in this file are mostly whitespace due to grouping tests together. I recommend reviewing with `?w=1`.